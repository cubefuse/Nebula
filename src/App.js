import React from "react";
import { connect } from "react-redux";
import { createIpfsInstance } from "./redux/ipfs.redux";
import AppHeader from "./components/common/AppHeader";
import AppFooter from "./components/common/AppFooter";
import BreadcrumbsBar from "./components/common/BreadcrumbsBar";
import { HashRouter as Router, Route } from "react-router-dom";
import DownloadPageContainer from "./containers/DownloadPageContainer";
import FileExplorerContainer from "./containers/FileExplorerContainer";
import { Layout } from "antd";

const { Content } = Layout;

class App extends React.Component {
  constructor(props) {
    super(props);

    props.dispatch(createIpfsInstance());
  }

  render() {
    return (
      <Layout style={{ minHeight: "100vh" }}>
        <AppHeader />
        <Content style={{ padding: "0 50px" }}>
          <BreadcrumbsBar />
          <Layout style={{ padding: "24px 0", background: "#fff" }}>
            <Router>
              <div>
                <Route exact path="/" component={FileExplorerContainer} />
                <Route path="/get" component={DownloadPageContainer} />
              </div>
            </Router>
          </Layout>
        </Content>
        <AppFooter />
      </Layout>
    );
  }
}

export default connect()(App);
