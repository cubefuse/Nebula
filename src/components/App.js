import React from "react";
import { Layout } from "antd";
import { Switch } from "react-router";
import AppHeader from "./common/AppHeader";
import AppFooter from "./common/AppFooter";
import BreadcrumbsBar from "./common/BreadcrumbsBar";
import { Redirect, Route } from "react-router-dom";
import DownloadPageContainer from "../containers/DownloadPageContainer";
import FileExplorerContainer from "../containers/FileExplorerContainer";
import Error404 from "./statuses/Error404";
import Loading from "./statuses/Loading";
import Error500 from "./statuses/Error500";

const { Content } = Layout;

class App extends React.Component {
  renderRoute() {
    const routes = (
      <Switch>
        <Route exact path="/" component={FileExplorerContainer} />
        <Route path="/get/:hash" component={DownloadPageContainer} />
        <Route path="/404" component={Error404} />
        <Redirect to="/404" />
      </Switch>
    );

    switch (this.props.ipfs.connState) {
      case "offline":
        return <Loading message="Loading the app" />;
      case "connecting":
        return <Loading message="Connecting to IPFS Network" />;
      case "connected":
        return routes;
      case "failed":
        return <Error500 error="Failed to connect to IPFS." />;
      default:
        return <Error500 error="Unknown IPFS State." />;
    }
  }

  componentDidMount() {
    this.props.createIpfsInstance();
  }

  render() {
    return (
      <Layout style={{ minHeight: "100vh" }}>
        <AppHeader />
        <Content style={{ padding: "0 24px" }}>
          <BreadcrumbsBar />
          <Layout
            style={{ padding: "24px 0", background: "#fff", height: "100%" }}
          >
            {this.renderRoute()}
          </Layout>
        </Content>
        <AppFooter />
      </Layout>
    );
  }
}

export default App;
