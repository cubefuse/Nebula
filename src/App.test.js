import React from "react";
import 'jest-dom/extend-expect'
import App from "./App";
import TestHelpers from "./util/test-helpers";


test.skip('can render with redux with defaults', () => {
  const node = TestHelpers.renderWithRedux(<App />);
  expect(node).toMatchSnapshot();
});