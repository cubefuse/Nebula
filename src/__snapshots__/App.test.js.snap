// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can render with redux with defaults 1`] = `
Object {
  "container": <div>
    <div
      class="ant-layout"
      style="min-height: 100vh;"
    >
      <div
        class="header ant-layout-header"
      >
        <div
          style="color: white;"
        >
          <span
            style="font-size: x-large; padding-right: 5px;"
          >
            <strong>
              Nebula
            </strong>
          </span>
          <span
            style="font-size: large; padding-right: 10px;"
          >
            Web UI
          </span>
          <span
            style="font-size: small;"
          >
            POC-01
          </span>
        </div>
      </div>
      <div
        class="ant-layout-content"
        style="padding: 0px 50px;"
      >
        <div
          class="ant-breadcrumb"
          style="margin: 16px 0px;"
        >
          <span>
            <span
              class="ant-breadcrumb-link"
            >
              Home
            </span>
            <span
              class="ant-breadcrumb-separator"
            >
              /
            </span>
          </span>
          <span>
            <span
              class="ant-breadcrumb-link"
            >
              My Files
            </span>
            <span
              class="ant-breadcrumb-separator"
            >
              /
            </span>
          </span>
        </div>
        <div
          class="ant-layout ant-layout-has-sider"
          style="padding: 24px 0px; background: rgb(255, 255, 255);"
        >
          <div>
            <div
              class="ant-layout-content"
              style="padding: 0px 24px; min-height: 280px;"
            >
              <div
                class="ant-layout-sider ant-layout-sider-dark"
                style="background: rgb(255, 255, 255); max-width: 200px; min-width: 200px; width: 200px;"
              >
                <div
                  class="ant-layout-sider-children"
                >
                  <ul
                    class="ant-menu ant-menu-light ant-menu-root ant-menu-inline"
                    role="menu"
                    style="height: 100%;"
                  >
                    <li
                      class="ant-menu-item ant-menu-item-selected"
                      role="menuitem"
                      style="padding-left: 24px;"
                    >
                      My Files
                    </li>
                    <li
                      class="ant-menu-item"
                      role="menuitem"
                      style="padding-left: 24px;"
                    >
                      Trash
                    </li>
                    <li
                      class="ant-menu-item"
                      role="menuitem"
                      style="padding-left: 24px;"
                    >
                      Shared
                    </li>
                  </ul>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-layout-footer"
        style="text-align: center;"
      >
        Copyright Â© 2018 Cubefuse. Distributed under MIT license.
      </div>
    </div>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "store": Object {
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
    Symbol(observable): [Function],
  },
  "unmount": [Function],
}
`;
